applications/adv_networking_bench/cpp/testing/adv_networking_test_harness.py

```diff
@@ -0,0 +1,243 @@
+    logger.debug("Validating benchmark results")
+
+    success = True
+    # Check for errors in the log
+    if "[error]" in log:
+        logger.error("Errors found in benchmark output")
+        success = False
+
+    # Parse benchmark results
+    transmit_pkts = int(re.search(r"Port 0.*?Transmit packets:\s+(\d+)", log, re.DOTALL).group(1))
+    received_pkts = int(re.search(r"Port 1.*Received packets:\s+(\d+)", log, re.DOTALL).group(1))
+    received_bytes = int(re.search(r"Port 1.*Received bytes:\s+(\d+)", log, re.DOTALL).group(1))
+    missed_pkts = int(re.search(r"Port 1.*Missed packets:\s+(\d+)", log, re.DOTALL).group(1))
+    errored_pkts = int(re.search(r"Port 1.*Errored packets:\s+(\d+)", log, re.DOTALL).group(1))
+    exec_time = float(re.search(r"TOTAL EXECUTION TIME OF SCHEDULER : (\d+\.\d+) ms", log).group(1))
+
+    logger.debug("Transmit packets:", transmit_pkts)
+    logger.debug("Received packets:", received_pkts)
+    logger.debug("Received bytes:", received_bytes)
+    logger.debug("Missed packets:", missed_pkts)
+    logger.debug("Errored packets:", errored_pkts)
+    logger.debug("Exec time:", exec_time)
+
+    # Check for missed packets
+    # NOTE: default expects no missed packets, however current app isn't configured
+    #   to consistently start tx or rx first, and there could be some dropped packets
+    #   at startup based on the order of things. Adding a slightly higher threshold
+    #   based on the test duration is reasonable until that is addressed.
+    missed_pkts_percent = (missed_pkts / transmit_pkts) * 100
+    if missed_pkts_percent > missed_pkts_threshold:
+        logger.error(
+            f"Missed packets: {missed_pkts} ({missed_pkts_percent:.2f}%, over {missed_pkts_threshold}% threshold ❌)"
+        )
+        success = False
+    else:
+        logger.info(
+            f"Missed packets: {missed_pkts} ({missed_pkts_percent:.2f}%, below {missed_pkts_threshold}% threshold ✅)"
+        )
+
+    # Check for errored packets
+    errored_pkts_percent = (errored_pkts / transmit_pkts) * 100
+    if errored_pkts_percent > error_pkts_threshold:
+        logger.error(
+            f"Errored packets: {errored_pkts} ({errored_pkts_percent:.2f}%, over {error_pkts_threshold}% threshold ❌)"
+        )
+        success = False
+    else:
+        logger.info(
+            f"Errored packets: {errored_pkts} ({errored_pkts_percent:.2f}%, below {error_pkts_threshold}% threshold ✅)"
+        )
+
+    # Calculate average throughput
+    # NOTE: This is an approximate and conservative heuristic, since:
+    #   - exec_time includes the overhead from startup and shutdown of the holoscan graph
+    #   - exec_time includes times when Rx might be up but Tx isn't yet sending
+    #   An ideal test would trim the values at the beginning and end. A better tool
+    #   is `mlnx_perf -i <interface>` but that would complicate this test wrapper.
+    avg_throughput = (received_bytes * 8) / (exec_time / 1000) / 1e9
```

Comment:
Do we know how accurate this is going to be?

Reply:
It depends on how long you run it. so far for a 20s run gives me 93 Gbps average with this heuristic, when I look at `mlnx_perf` I am pretty consistently at 98 Gbps during the run, excluding beginning and end.

The way I see it, this is good enough for a first step, we're more interested to see if for some reason our perf goes way lower. And if we want to improve, we could spin-off `mlnx_perf` in parallel and parse that, trimming some, but it's more effort that I don't think is worth it for what we're trying to achieve (which is not an accurate benchmark, but a rough quality check).